node/pkg/vaa/structs_test.go

```diff
@@ -123,3 +129,179 @@ func TestChainId_String(t *testing.T) {
+		Sequence:         uint64(1),
+		ConsistencyLevel: uint8(32),
+		EmitterChain:     ChainIDSolana,
+		EmitterAddress:   governanceEmitter,
+		Payload:          payload,
+	}
+
+	expected := common.HexToHash("4fae136bb1fd782fe1b5180ba735cdc83bcece3f9b7fd0e5e35300a61c8acd8f")
+	assert.Equal(t, vaa.SigningMsg(), expected)
+}
+
+func TestMessageID(t *testing.T) {
+	var payload = []byte{97, 97, 97, 97, 97, 97}
+	var governanceEmitter = Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}
+
+	vaa := &VAA{
+		Version:          uint8(1),
+		GuardianSetIndex: uint32(1),
+		Signatures:       nil,
+		Timestamp:        time.Unix(0, 0),
+		Nonce:            uint32(1),
+		Sequence:         uint64(1),
+		ConsistencyLevel: uint8(32),
+		EmitterChain:     ChainIDSolana,
+		EmitterAddress:   governanceEmitter,
+		Payload:          payload,
+	}
+
+	expected := "1/0000000000000000000000000000000000000000000000000000000000000004/1"
+	assert.Equal(t, vaa.MessageID(), expected)
+}
+
+func TestHexDigest(t *testing.T) {
+	var payload = []byte{97, 97, 97, 97, 97, 97}
+	var governanceEmitter = Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}
+
+	vaa := &VAA{
+		Version:          uint8(1),
+		GuardianSetIndex: uint32(1),
+		Signatures:       nil,
+		Timestamp:        time.Unix(0, 0),
+		Nonce:            uint32(1),
+		Sequence:         uint64(1),
+		ConsistencyLevel: uint8(32),
+		EmitterChain:     ChainIDSolana,
+		EmitterAddress:   governanceEmitter,
+		Payload:          payload,
+	}
+
+	expected := "4fae136bb1fd782fe1b5180ba735cdc83bcece3f9b7fd0e5e35300a61c8acd8f"
+	assert.Equal(t, vaa.HexDigest(), expected)
+}
+
+func TestVerifySignatures(t *testing.T) {
+	var payload = []byte{97, 97, 97, 97, 97, 97}
+	var governanceEmitter = Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}
+
+	vaa := &VAA{
+		Version:          uint8(1),
+		GuardianSetIndex: uint32(1),
+		Signatures:       nil,
+		Timestamp:        time.Unix(0, 0),
+		Nonce:            uint32(1),
+		Sequence:         uint64(1),
+		ConsistencyLevel: uint8(32),
+		EmitterChain:     ChainIDSolana,
+		EmitterAddress:   governanceEmitter,
+		Payload:          payload,
+	}
+
+	// Generate a random private key to sign with
+	privKey, _ := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	assert.Nil(t, vaa.Signatures)
+
+	// Add a signature and make sure it's added
+	vaa.AddSignature(privKey, 0)
+	assert.Equal(t, len(vaa.Signatures), 1)
+
+	// Generate a public key to compare to from our private key
+	h := vaa.SigningMsg()
+	pubKey, err := crypto.Ecrecover(h.Bytes(), vaa.Signatures[0].Signature[:])
+	assert.Nil(t, err)
+	assert.NotNil(t, pubKey)
+
+	// Translate that public key back to an address
+	addr := common.BytesToAddress(crypto.Keccak256(pubKey[1:])[12:])
+
+	// Make sure that it verifies
+	assert.True(t, vaa.VerifySignatures([]common.Address{addr}))
```

Comment:
Also this tests all of the positive cases, what what about the negative cases where there are errors? We probably want to cover those with a followup to this later on. Error code paths are always the least often used and most likely to break things.

Reply:
I have another PR that goes into much greater depth on this topic.  This PR is largely just to establish a base from which to build an extensive suite of tests on top.